fineract-core/src/main/java/org/apache/fineract/portfolio/savings/data/SavingsAccountTransactionData.java

```diff
@@ -117,7 +117,7 @@ public static SavingsAccountTransactionData interestPosting(final SavingsAccount
                 savingsAccountTransactionType.getValue().longValue(), savingsAccountTransactionType.getCode(),
                 savingsAccountTransactionType.getValue().toString());
         return new SavingsAccountTransactionData(amount.getAmount(), date, savingsAccount.getId(), savingsAccountTransactionEnumData,
-                isReversed, null, isManualTransaction, lienTransaction);
+                isReversed, null, isManualTransaction, lienTransaction, date);
```

Comment:
This looks wrong to me. We should not set the submitted on date by the transaction date.

Reply:
This is the period end due date of an interest posting period, which means that the transactionDate and the submittedOnDate are the same (or will be the same when the interest will be posted)

Reply:
Is that mean the transaction date is end of month (31th July) or start of month (1st of Aug)?

Reply:
i dont think it should be left empty. it should be set the actual business date. However in the above example the transaction date and submitted on date cannot be the same. So i dont think we should set the submitted on date based on the transaction date.

Reply:
AFAIK the job picks the same day interest dues, so your example would not happen. But if there is no business date set, then it can happen that the job runs days later and still posts interest from the past and in this case you are right that the submittedOnDate should not be set to the transaction date.
@Mk9894 please change posting submitted dates to getBusinessDate

Reply:
if it is used for template purpose and the submitted on date is not null, thats not a big problem, however whenever this data object is used as a result of an API call or as the data object which will be used to create savings transaction, the submitted on date should be correct!

Reply:
This one was not the question here

Reply:
i might misunderstood the `the submittedOnDate would be better to keep null` part... mea culpa